ci/Jenkinsfiles/export.groovy

```diff
@@ -0,0 +1,329 @@
+          """
+          if (params.NUXEO_VERSION.isEmpty()) {
+            currentBuild.result = 'ABORTED';
+            def message = 'NUXEO_VERSION parameters is mandatory'
+            currentBuild.description = "${message}"
+            echo "${message}"
+            error(currentBuild.description)
+          }
+        }
+      }
+    }
+
+    stage('Set Labels') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Set Kubernetes resource labels
+          ----------------------------------------
+          """
+          echo "Set label 'branch: export' on pod ${NODE_NAME}"
+          sh """
+            kubectl label pods ${NODE_NAME} branch=export
+          """
+        }
+      }
+    }
+
+    stage('Build Docker Image') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Build Explorer Export Docker Image
+          ----------------------------------------
+          Image tag: ${VERSION}
+          Registry: ${DOCKER_REGISTRY}
+          """
+          script {
+            def moduleDir = 'docker/nuxeo-explorer-export-docker'
+            def exportCredsId = "${params.DOWNLOAD_PACKAGES_FROM_PROD ? 'explorer-connect-prod-clid' : 'explorer-connect-preprod-clid'}"
+            // push images to the Jenkins X internal Docker registry
+            withCredentials([string(credentialsId: exportCredsId, variable: 'CONNECT_EXPORT_CLID')]) {
+              sh """
+                envsubst < ${moduleDir}/skaffold.yaml > ${moduleDir}/skaffold.yaml~gen
+                skaffold build -f ${moduleDir}/skaffold.yaml~gen
+                # waiting skaffold + kaniko + container-stucture-tests issue
+                #  see https://github.com/GoogleContainerTools/skaffold/issues/3907
+                docker pull ${DOCKER_REGISTRY}/nuxeo/nuxeo-explorer-export:${VERSION}
+              """
+            }
+          }
+        }
+      }
+    }
+
+    stage('Deploy Export Preview and Perform Export') {
+      steps {
+        container('maven') {
+          dir('helm/export') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute docker image names and versions
+            sh """
+              echo ${CONNECT_EXPLORER_URL}
+              echo ${NUXEO_EXPLORER_PACKAGE}
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              try {
+                boolean nsExists = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+                // Only used with jx preview on pr branches
```

Comment:
There's no preview on PR branches right?

Reply:
cleaned this comment up (as well as unused noCommentOpt)